File: app/src/main/java/com/eccentricyan/pohe/presentation/article/ArticlesChangeEvent.java
package com.eccentricyan.pohe.presentation.article;

import com.eccentricyan.pohe.domain.entity.Article;

import java.util.HashMap;
import java.util.List;

/**
 * Created by shiyanhui on 2017/04/15.
 */

public class ArticlesChangeEvent {
    public final HashMap<String, List<Article>> map = new HashMap<>();

    public final boolean refresh;
    public ArticlesChangeEvent(List<Article> articles, boolean refresh, String category) {
        this.map.put(category, articles);
        this.refresh = refresh;
    }
}

File: app/src/main/java/com/eccentricyan/pohe/presentation/article/ArticlesChangeEvent.kt
package com.eccentricyan.pohe.presentation.article

import com.eccentricyan.pohe.domain.entity.Article

import java.util.HashMap

/**
 * Created by shiyanhui on 2017/04/15.
 */

class ArticlesChangeEvent(articles: List<Article>, val refresh: Boolean, category: String) {
    val map = HashMap<String, List<Article>>()

    init {
        this.map[category] = articles
    }
}
